app/uk/gov/hmrc/gform/sdes/SdesService.scala

```diff
@@ -139,15 +148,44 @@ class SdesService(
     showBeforeAt: Option[Boolean],
     destination: Option[SdesDestination]
   ): Future[SdesSubmissionPageData] = {
+    val skip = page * pageSize
+    doSearch(Some((skip, pageSize)), processed, formTemplateId, status, showBeforeAt, destination, None)
+  }
+
+  override def searchAll(
+    processed: Option[Boolean],
+    formTemplateId: Option[FormTemplateId],
+    status: Option[NotificationStatus],
+    showBeforeAt: Option[Boolean],
+    destination: Option[SdesDestination],
+    showBeforeLastUpdatedAt: Option[Int]
+  ): Future[SdesSubmissionPageData] =
+    doSearch(None, processed, formTemplateId, status, showBeforeAt, destination, showBeforeLastUpdatedAt)
+
+  private def doSearch(
+    maybeSkipAndPageSize: Option[(Int, Int)],
+    processed: Option[Boolean],
+    formTemplateId: Option[FormTemplateId],
+    status: Option[NotificationStatus],
+    showBeforeAt: Option[Boolean],
+    destination: Option[SdesDestination],
+    showBeforeLastUpdatedAt: Option[Int]
+  ): Future[SdesSubmissionPageData] = {
 
     val queryByTemplateId =
       formTemplateId.fold(exists("_id"))(t => equal("formTemplateId", t.value))
     val queryByProcessed =
       processed.fold(queryByTemplateId)(p => Filters.and(equal("isProcessed", p), queryByTemplateId))
     val queryByStatus =
       status.fold(queryByProcessed)(s => Filters.and(equal("status", fromName(s)), queryByProcessed))
+
+    val queryByLastUpdated =
+      showBeforeLastUpdatedAt.fold(queryByStatus)(d =>
+        Filters.and(lt("lastUpdated", LocalDateTime.now().minusHours(d.toLong)), queryByStatus)
+      )
+
     val queryByDestination =
-      destination.fold(queryByStatus)(d =>
+      destination.fold(queryByLastUpdated)(d =>
         Filters.and(equal("destination", SdesDestination.fromName(d)), queryByStatus)
       )
     val query = if (showBeforeAt.getOrElse(false)) {
```

Comment:
Can we refactor this? This is starting to become unmaintainable

We can define conditions one by one and combine them all later in one go

```
// pseudocode:
val queryByProcessed: Option[Bson] = processed.map(p => equal("isProcessed", p))
val queryByStatus: Option[Bson] = status.map(s => equal("status", fromName(s)))
val queryByLastUpdated: Option[Bson] = showBeforeLastUpdatedAt.map(d => lt("lastUpdated", LocalDateTime.now().minusHours(d.toLong)))
val queryByDestination: Option[Bson] = destination.map(d => equal("destination", SdesDestination.fromName(d)))
....
val conds: List[Bson] =
      queryByTemplateId :: List(queryByProcessed, queryByStatus, queryByLastUpdated, queryByDestination).collect {
        case Some(cond) => cond
      }
val query = Filters.and(conds: _*)
```